modules/@apostrophecms/soft-redirect/index.js

```diff
@@ -37,7 +37,9 @@ module.exports = {
     return {
       '@apostrophecms/page:notFound': {
         async notFoundRedirect(req) {
-          const doc = await self.apos.doc.find(req, { historicUrls: { $in: [ req.url ] } }).sort({ updatedAt: -1 }).toObject();
+          const url = new URL(`${req.baseUrl}${req.url}`);
```

Comment:
the issue was that `req.url` contains the query string, whereas `historicUrls` do not

Reply:
I see why you need to arrive at just the pathname, but please just remove `?.*$` from `req.url` to determine the pathname, rather than feeding it through the full URL parser. Otherwise I have a suspicion that we could hit edge cases where req.baseUrl doesn't parse cleanly.